src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -1,28 +1,76 @@
 package com.github.sdonkov;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jol.info.GraphLayout;
 
 
 import java.io.IOException;
 import java.net.URISyntaxException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.TimeUnit;
 
 import static org.junit.jupiter.api.Assertions.*;
 
-
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Fork(value = 1, warmups = 1)
+@State(Scope.Benchmark)
 public class PrefixTreeTest {
 
     private static List<String> words;
     private static PrefixTree prefixTree;
+    private final static Logger LOGGER = LogManager.getLogger(PrefixTree.class);
 
     @BeforeAll
     static void setUp() throws IOException, URISyntaxException {
         prefixTree = new PrefixTree();
-        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/tests.txt").toURI()));
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
+        words.forEach(word -> prefixTree.add(word));
+    }
+
+    @Test
+    public void launchBenchmark() throws RunnerException {
+        Options opt = new OptionsBuilder()
+                .include(PrefixTreeTest.class.getSimpleName())
+                .build();
+        new Runner(opt).run();
+    }
+
+    @Benchmark
+    public void benchmarkPrefixTree() throws URISyntaxException, IOException {
+        prefixTree = new PrefixTree();
+        words = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/dictionary.txt").toURI()));
         words.forEach(word -> prefixTree.add(word));
+        LOGGER.trace(GraphLayout.parseInstance(prefixTree).totalSize() + " prefixTree");
```

Comment:
This is not what we want from the microbenchmark.
We need to know how fast the structure works. Currently we have two methods exposed as functionality - `add` and `contains`. I say we test only `contains`. The test should be something like:
0. Setup the test subject by adding all words from the dictionary. Pick randomly 10 words from the dictionary. This step should be performed prior to running the benchmark.
1. Test how fast does the structure checks whether all the words are contained and repeat the test many times to get an average. The second part is provided by the microbenchmark framework.

Outside of the benchmark, it'd be very useful to print how much memory each tested structure consumes. There's no need to check many times.